protocol/messenger_chats.go

```diff
@@ -575,3 +575,45 @@ func (m *Messenger) clearHistory(id string) (*MessengerResponse, error) {
 	response.AddChat(chat)
 	return response, nil
 }
+
+func (m *Messenger) ShareImageMessage(request *requests.ShareImageMessage) (*MessengerResponse, error) {
+	if err := request.Validate(); err != nil {
+		return nil, err
+	}
+	response := &MessengerResponse{}
+
+	msg, err := m.persistence.MessageByID(request.MessageID)
+	if err != nil {
+		return nil, err
+	}
+
+	var messages []*common.Message
+	for _, pk := range request.Users {
+		message := &common.Message{}
+		message.ChatId = pk.String()
+		message.Payload = msg.Payload
+		message.ContentType = protobuf.ChatMessage_ContentType(request.ContentType)
```

Comment:
Do we not know the content type based on the fact that you are calling "ShareImageMessage"?